PKGBUILD

```diff
@@ -0,0 +1,43 @@
+# Maintainer: Cosmin Gabriel Georgescu <cosmingg2013@gmail.com>
+pkgname="duckscript"
+pkgver=0.8.20
+pkgrel=1
+pkgdesc="Simple, extendable and embeddable scripting language."
+
+# https://github.com/sagiegurari/duckscript#installation-binary-release
+arch=("x86_64")
+
+url="https://sagiegurari.github.io/duckscript/"
+
+# https://github.com/sagiegurari/duckscript/blob/master/LICENSE
+license=("Apache")
+
+makedepends=("cargo")
+
+source=("$pkgname-$pkgver.tar.gz::https://github.com/sagiegurari/duckscript/archive/refs/tags/$pkgver.tar.gz")
+
+sha512sums=("7ffe4ad2d1bb54753d7c262981bf31394404dd71a75919188a5f658de39b9b9215d057da382e220433d866369417e74feaef3ab4ac80df5214b046284f5f840d")
+
+b2sums=("869859f0d7e21abf1f3e1041e6f7056c25d4e6341bab19788c95c609da60cff5add66ec113aa1f2ff9557596537728a6255ec65c280668d9d3e8e785d042f616")
+
+prepare(){
+	export RUSTUP_TOOLCHAIN=stable
+	cargo fetch --locked
+}
+
+build(){
+	cd "$pkgname-$pkgver"
+	export RUSTUP_TOOLCHAIN=stable
+	export CARGO_TARGET_DIR=target
+	cargo build --frozen --workspace --release --all-features
+}
+
+check(){
+	cd "$pkgname-$pkgver"
+	export RUSTUP_TOOLCHAIN=stable
+	cargo test --frozen --workspace --all-features
```

Comment:
some tests are not thread safe and some are.
duckscript by default uses cargo-make which splits the run to single thread and multi thread.
single thread tests are by default ignored (which is why your test command works), so what cargo-make does is 
1. run cargo test on everything so its multi thread and single thread tests are ignored
2. sets RUST_TEST_THREADS=1 and runs cargo test with --ignored

so if you want to run ALL tests, thats what you got to do.
if you want to run with partial coverage, you can keep what you did.

wdyt?

Reply:
Sorry about the delay; had some finals.
I would go with running the full test suite.

Reply:
I've pushed the commit that should address this. So now,
`cargo test --frozen --workspace --all-features | grep ignored` will report 
>test result: ok. 179 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.01s
test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
test lib_test::test_scripts ... ignored
test sdk::std::env::set_current_directory::mod_test::run_no_arguments ... ignored
test sdk::std::env::set_current_directory::mod_test::run_specific_directory ... ignored
test sdk::std::env::set_current_directory::mod_test::run_to_directory_does_not_exist ... ignored
test result: ok. 823 passed; 0 failed; 4 ignored; 0 measured; 0 filtered out; finished in 0.92s
test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s 

and `cargo test --frozen --workspace --all-features -- --ignored --test-threads=1 | grep -w test` will report the remaining tests
>test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 179 filtered out; finished in 0.00s
test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
test lib_test::test_scripts ... hello world
test sdk::std::env::set_current_directory::mod_test::run_no_arguments ... ok
test sdk::std::env::set_current_directory::mod_test::run_specific_directory ... ok
test sdk::std::env::set_current_directory::mod_test::run_to_directory_does_not_exist ... ok
test result: ok. 4 passed; 0 failed; 0 ignored; 0 measured; 823 filtered out; finished in 6.77s
test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

In total, 1006 tests were run.
Let me know if I missed anything.